packages/swap/src/offer.ts

```diff
@@ -4,8 +4,13 @@ import { Transaction } from '@bitcoin-computer/nakamotojs'
 const { Contract } = await import('@bitcoin-computer/lib')
 
 export class Offer extends Contract {
-  constructor(owner: string, url: string, json: string) {
-    super({ _owners: [owner], _url: url, json })
+  txHex: string
+  constructor(owner: string, url: string, txHex?: string) {
```

Comment:
We should remove the owner here. If no owner is specified the Bitcoin Computer uses the public key of the computer object that broadcasts the transaction.

The difference is that if the owner is passed as a parameter, it is added to the metadata and we end up storing it twice (in the metadata and the output script). If we omit it, it only occurs in the output script. That makes using the contract a little cheaper...